### DISCUSSION TITLE
Performing tasks after responding to request in JS hooks

### DISCUSSION THREAD

**Discussion by JunoNgx** [pocketbase/pocketbase#4581] [Q&A] [answered]
Context: in my app, the user enters a random content, and if the content is detected to be a valid url, I want Pocketbase to try retrieving the title and absolute favicon url of this page.

Important requirement: I also need this request to be responded to immediately (that the record has been created), without letting the requester wait for the meta-data to be fetched.

I attempted to implement this by intercepting the hook `onRecordAfterCreateRequest`.

In Go, I managed to achieve the immediate response this by creating a goroutine and proceeding to synchronously `return`:
```
if (isValidUrl) {
    go TryFetchTitleAndFavicon()
    return nil
}
```

Is there anyway to achieve a similar effect in JS hooks?

**Comment by ganigeorgiev** [comment, marked as answer]
All JSVM operations are blocking within their handler function/scope and there is no support for `go fn()` equivalent or any concurrent operation in general within a single hook, if that is what you are asking (_you can find an overview of how the integration works in https://github.com/pocketbase/pocketbase/discussions/2825#discussioncomment-6376210_).

If you really have to use the JSVM for this use case, an alternative could be to have a helper "jobs/queue" collection and create a new record for it from inside the `onRecordAfterCreateRequest` (_make sure to specify the collection name with the hook to prevent infinite loop_). Then you can [schedule a cron job](https://pocketbase.io/docs/js-jobs-scheduling/) and process the entry from the "jobs/queue" collection (_note that the min interval granularity is 1 minute_).
